sycl/source/detail/scheduler/commands.cpp

```diff
@@ -1812,9 +1839,27 @@ cl_int ExecCGCommand::enqueueImp() {
     for (ArgDesc &Arg : ExecKernel->MArgs) {
       switch (Arg.MType) {
       case kernel_param_kind_t::kind_accessor: {
+// TODO ESIMD Currently all accessors are treated as ESIMD under corresponding
+// compiler option enabling the macro below. Eventually ESIMD kernels and usual
+// kernels must co-exist and there must be a mechanism for distinguishing usual
+// and ESIMD accessors.
+#ifndef __SYCL_EXPLICIT_SIMD__
```

Comment:
This macro won't work because this file is a part of libsycl.so. Suggest adding  ESIMD flag to  AccessorImplHost and check it dynamically where needed.

Reply:
Right, thanks. This was a last-minute change for the open-source variant, could not really test it at this point.